fitbenchmarking/controllers/base_matlab_controller.py

```diff
@@ -0,0 +1,78 @@
+"""
+Implements base class for the matlab fitting software controllers.
+"""
+
+try:
+    from tempfile import TemporaryDirectory
+except ImportError:
+    from backports.tempfile import TemporaryDirectory
+import os
+try:
+    import dill
+    import_success = True
+except ImportError:
+    import_success = False
+
+from fitbenchmarking.controllers.base_controller import Controller
+from fitbenchmarking.utils.exceptions import MissingSoftwareError
+
+
+class BaseMatlabController(Controller):
+    """
+    Base class for matlab fitting software controllers
+    """
+
+    def __init__(self, cost_func):
+        """
+        Initialise anything that is needed specifically for matlab
+        fitting software
+        """
+        super().__init__(cost_func)
+
+        if not import_success:
+            raise MissingSoftwareError('Requirements are missing for Matlab '
+                                       'fitting, module "dill" is reuqired.')
+
+        self.initial_params_mat = None
+
+    def setup(self):
+        raise NotImplementedError
+
+    def jacobian_information(self):
+        raise NotImplementedError
+
+    def fit(self):
+        raise NotImplementedError
+
+    def cleanup(self):
+        raise NotImplementedError
```

Comment:
I added these functions in because otherwise I got an 'abstract-method' pylint warning saying that the method is abstract in class 'Controller' but is not overridden. Do you know of any ways I could make pylint happy without including these functions?

Reply:
I couldn't remove pylint warnings using ABCMeta, so have instead made `BaseMatlabController` independent of `Controller`, meaning the matlab controllers now inherit from both the base matlab controller and base controller classes